src/components/NavBar.jsx

```diff
@@ -1,11 +1,27 @@
 import { Link } from 'react-router-dom'
+import { useState } from "react";
+
 const NavBar = () => {
-        return (
+
+const [topic, setTopic] = useState("");
+
+return (
+            <div>
         <nav>
             <Link to="/" ><button>Home</button>  </Link>
-            <Link to="/articles" > <button>Articles</button> </Link>
+            <Link to="/articles" > <button onClick={() => setTopic("")}>Articles</button> </Link>
             <Link to="/users" > <button>Users</button> </Link>
+            
+            
+            <Link to="/articles/topics/coding"> <button onClick={() => setTopic('coding')}>Coding</button> </Link>
+
+            <Link to="/articles/topics/football"><button onClick={() => setTopic('football') }>Football</button> </Link>
+
+            <Link to="/articles/topics/cooking"><button onClick={() => setTopic('cooking') }>Cooking</button></Link> 
         </nav>
+
```

Comment:
❗  PLEASE NOTE: you do not need the topic state at all nor any of the onclicks in this nav file because you're getting topic straight from the parametric endpoint and not even using the topic state (see React warnings) so you can take it out ❗